src/segments/unity.go

```diff
@@ -0,0 +1,105 @@
+package segments
+
+import (
+	"errors"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/jandedobbeleer/oh-my-posh/src/platform"
+	"github.com/jandedobbeleer/oh-my-posh/src/properties"
+	"github.com/jandedobbeleer/oh-my-posh/src/regex"
+)
+
+type Unity struct {
+	props properties.Properties
+	env   platform.Environment
+
+	UnityVersion  string
+	CSharpVersion string
+}
+
+func (u *Unity) Enabled() bool {
+	unityVersion, err := u.GetUnityVersion()
+	if err != nil {
+		u.env.Error(err)
+		return false
+	}
+	if len(unityVersion) == 0 {
+		return false
+	}
+	u.UnityVersion = unityVersion
+
+	csharpVersion, err := u.GetCSharpVersion()
+	if err != nil {
+		u.env.Error(err)
+	}
+	u.CSharpVersion = csharpVersion
+
+	return true
+}
+
+func (u *Unity) GetUnityVersion() (version string, err error) {
+	versionFilePath := filepath.Join("ProjectSettings", "ProjectVersion.txt")
```

Comment:
This will only work when you're inside the root folder, correct? If so, what we can also do is use `u.env.HasParentFilePath("ProjectSettings")` and then validate if have the correct file using `u.env.HasFiles("ProjectVersion.txt")`. For example:

```golang
projectDir, err := g.env.HasParentFilePath("ProjectSettings")
if err != nil {
    u.env.Debug("No ProjectSettings parent folder found")
	return
}

if !u.env.HasFilesInDir(projectDir.Path, "ProjectVersion.txt") {
    u.env.Debug("No ProjectVersion.txt file found")
	return
}

versionFilePath := filepath.Join(projectDir.Path, "ProjectVersion.txt")
versionFileText := u.env.FileContent(versionFilePath)
```

Reply:
Ah, yes. Good suggestion.